src/researchhub_comment/signals/rh_comment_signals.py

```diff
@@ -0,0 +1,186 @@
+from django.contrib.contenttypes.models import ContentType
+from django.core.files.base import ContentFile
+from django.db import transaction
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from discussion.models import (
+    Comment as LegacyComment,
+    Reply as LegacyReply,
+    Thread as LegacyThread,
+)
+from researchhub_comment.constants.rh_comment_content_types import QUILL_EDITOR
+from researchhub_comment.constants.rh_comment_migration_legacy_types import LEGACY_COMMENT, LEGACY_REPLY, LEGACY_THREAD
+from researchhub_comment.constants.rh_comment_thread_types import GENERIC_COMMENT
+from researchhub_comment.models import RhCommentThreadModel, RhCommentModel
+from utils import sentry
+
+
+@receiver(post_save, sender=LegacyThread, dispatch_uid='from_legacy_thread_to_rh_comment')
+def from_legacy_thread_to_rh_comment(sender, instance, created, **kwargs):
+    try:
+        upserter = instance.created_by
+        belonging_document = get_belonging_doc(instance)
+        belonging_thread = find_or_create_belonging_thread(belonging_document, upserter)
+        comment_content_file = ContentFile((instance.plain_text or "").encode())
+        legacy_thread_id = instance.id
+
+        migrated_thread_comment = RhCommentModel.objects.filter(
+            legacy_id=legacy_thread_id,
+            legacy_model_type=LEGACY_THREAD,
+        ).first() or RhCommentModel.objects.create(
+            comment_content_type=QUILL_EDITOR,
+            context_title=instance.context_title,
+            created_by=upserter,
+            created_date=instance.created_date,
+            legacy_id=legacy_thread_id,
+            legacy_model_type=LEGACY_THREAD,
+            thread=belonging_thread,
+            updated_by=upserter,
+        )
+        migrated_thread_comment.context_title = instance.context_title or ""
+        migrated_thread_comment.comment_content_src.save(
+            f"rh-comment-user-{upserter.id}-thread-{belonging_thread.id}-comment-{migrated_thread_comment.id}.txt",
+            comment_content_file
+        )
+        # intentionally querying DB to ensure that the instance was created properly
+        return RhCommentModel.objects.get(
+            legacy_id=legacy_thread_id,
+            legacy_model_type=LEGACY_THREAD,
+        )
```

Comment:
I think ```migrated_thread_comment.id``` works just as well. If the instance was not created properly, it wouldn't have a ID assigned to it. The same applies to the other signals down below.

Reply:
I also wanted the instances to be returned to for recursive call reasons. I would like to keep this.